src/main/java/io/confluent/idesidecar/restapi/resources/RestProxyResource.java

```diff
@@ -43,16 +64,78 @@ public void schemaRegistryClusterProxy(RoutingContext routingContext) {
   }
 
   private void handleClusterProxy(RoutingContext routingContext, ClusterProxyContext proxyContext) {
-    clusterProxyProcessor.process(proxyContext)
+    process(routingContext, clusterProxyProcessor, proxyContext);
+  }
+
+  private void handleRBACProxy(RoutingContext routingContext, ProxyContext proxyContext) {
+    process(routingContext, rbacProxyProcessor, proxyContext);
+  }
+
+  public record RBACRequest(
+      @NotNull String userPrincipal,
+      @NotNull Action[] actions
+  ) {
+    public record Action(
+        @NotNull String resourceType,
+        @NotNull String resourceName,
+        @NotNull String operation,
+        @NotNull Scope scope
+    ) {
+      public record Scope(
+          @NotNull Map<String, String> clusters,
+          @NotNull String[] path
+      ) {}
+    }
+  }
+
+  public record RBACResponse(
+      @NotNull String title
+  ) {}
+
+  @Route(
+      path = RBAC_RESOURCE_PATH,
+      methods = Route.HttpMethod.PUT,
+      produces = MediaType.APPLICATION_JSON,
+      consumes = MediaType.APPLICATION_JSON
+  )
+
+  @Blocking
+  @Operation(summary = "RBAC proxy route", description = "Proxy route for RBAC requests")
+  @RequestBody(
+      description = "RBAC request body",
+      required = true,
+      content = @Content(schema = @Schema(implementation = RBACRequest.class))
+  )
+  @APIResponses({
+    @APIResponse(
+        responseCode = "200",
+        description = "Successful response",
+        content = @Content(schema = @Schema(implementation = RBACResponse.class))
+    ),
+    @APIResponse(
+        responseCode = "401",
+        description = "Unauthorized",
+        content = @Content(schema = @Schema(implementation = RBACResponse.class))
+    ),
+    @APIResponse(
+        responseCode = "404",
+        description = "Not Found",
+        content = @Content(schema = @Schema(implementation = RBACResponse.class))
+    )
+  })
+  public void RBACProxyRoute(RoutingContext routingContext) {
```

Comment:
Let's use camelCase for method names.

```suggestion
  public void rbacProxyRoute(RoutingContext routingContext) {
```